### DISCUSSION TITLE
401 Unauthorized: [{"message":"Bad credentials",] despite generating token correctly

### DISCUSSION THREAD

**Discussion by npaulzsdemo3** [community/community#136460] [Apps, API and Webhooks]
### Select Topic Area

General

### Body

Hi,
I am using OAuth to generate an access token. However the token generated does not have the gho_ in the prefix due to which i doubt that the api is giving the above error.

Is this a known issue or some change ?

Thanks,
Narinder

**Comment by GabrielAlmeidag** [comment]
@npaulzsdemo3 🐱‍🚀

The issue you’re encountering with generating an OAuth token that lacks the gho_ prefix and subsequently receiving a 401 Unauthorized: {"message":"Bad credentials"} error could stem from several factors, rather than being a known problem or recent change in the API.

First, tokens from GitHub typically begin with ghp_ for personal access tokens or gho_ for OAuth tokens. If your token lacks this prefix, it might indicate that it wasn’t generated correctly or that an outdated or incorrect authentication process is being used. Secondly, ensure that the token was generated with the correct scopes required for the API endpoint you’re trying to access. Missing necessary scopes can cause the API to reject the token with a 401 Unauthorized error.

Additionally, confirm that you’re sending the token in the Authorization header in the correct format, Bearer <token>. Any mistake in formatting or token transmission could result in “Bad credentials.” If you’re accessing a private repository or organization, check that the token has sufficient permissions to access these resources. Furthermore, tokens can expire or be revoked, so ensure that the token in use is still valid. It’s also advisable to consult the [GitHub API documentation](https://docs.github.com/en/rest/overview/other-authentication-methods) to see if there have been any recent changes in how tokens should be generated or used.

If all these aspects are correctly configured and the issue persists, consider generating a new token and testing again, or reach out to GitHub support for further assistance.
